server/message.go

```diff
@@ -0,0 +1,57 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+var (
+	ErrorMalformedClientMessage = errors.New("[coniks] Malformed client request")
+	ErrorNameExisted            = errors.New("[coniks] Registering identity is already registered")
+	ErrorInternalServer         = errors.New("[coniks] Internal server error")
+)
+
+const (
+	RegistrationType = "registration"
+)
+
+type Request struct {
+	Type    string
+	Request interface{}
+}
+
+type RegistrationRequest struct {
+	Username               string `json:"username"`
+	Key                    string `json:"key"`
+	KeyType                string `json:"key_type,omitempty"`
+	AllowUnsignedKeychange bool   `json:"allow_unsigned_key_change,omitempty"`
+	AllowPublicLookup      bool   `json:"allow_public_lookup,omitempty"`
+}
+
+type Response interface{}
+
+type ErrorResponse struct {
+	Error string
+}
+
+type RegistrationResponse struct {
+	TempraryBinding string `json:"tb"`
+}
+
+func NewErrorResponse(err error) *ErrorResponse {
+	return &ErrorResponse{Error: err.Error()}
+}
+
+func (reg *RegistrationRequest) serialize() []byte {
+	// serialize scheme:
```

Comment:
> But we already have ``message.go``. I think we can keep that name and move ``serialize()`` to something like ``messagekv.go``

I meant rename ``serialization.go`` to ``messaging.go`` since it contains functions for messaging between clients and the server. ``messagekv.go`` doesn't make so much sense to me unless the server needs an interface to store things like ``RegistrationRequest``s in the DB. I'm not sure this is necessary. Do you think the server should store all the requests it receives from the client in the DB?

> All ``***kv.go`` then go to a subpackage, e.g. ``kv`` package within ``server/kv`` directory. What do you think? 

This still sounds fine to me if the server needs additional interface(s) to the DB that aren't covered by the ``kv`` package we're going to introduce in the PR for #4. But as I mentioned above, I'm not sure it will actually need these.